client/src/util/testScheduler.ts

```diff
@@ -17,10 +17,26 @@ export interface CourseSection {
     endtime: string;
 }
 
-export const generateSchedules = (courses: CourseSection[], restrictedDays: number[] = []): CourseSection[][] => {
+export const generateSchedules = (
+    courses: CourseSection[],
+    restrictedDays: number[] = [],
+    selectedSections: string[] = []): CourseSection[][] => {
     const lectures = filterLectures(courses);
     const schedules = generateCourseScheduleOnlyLectures(lectures, restrictedDays);
-    return schedules;
+    return filterSchedules(schedules, selectedSections);
+};
+
+/**
+ * Given an array of schedules, filter out schedules that contain selected sections
```

Comment:
Just to clarify this wording, this function is basically making sure every schedule has all the selected sections right? Like if I lock CPSC 310 103, we're only including schedules that have this section.  When I read `filter out schedules` I understood it as `remove schedules that contain selected sections` - but the opposite is happening and we are actually keeping the schedules that are selected.

Reply:
Oh yeah I was half asleep when I was writing this comment, will update accordingly!